weblate/trans/models/unit.py

```diff
@@ -200,7 +200,13 @@ def search(self, query, **context) -> UnitQuerySet:
         """High level wrapper for searching."""
         from weblate.utils.search import parse_query
 
-        result = self.filter(parse_query(query, **context))
+        result = self
+        if "labels_count" in query:
+            result = result.annotate(
+                labels_count=Count("source_unit__labels") + Count("labels")
+            )
+
+        result = result.filter(parse_query(query, **context))
```

Comment:
The whole filtering should be implemented in `weblate/utils/search.py`. It is used from other places as well.